rocketbelt/components/buttons/rocketbelt.buttons-dropdown.js

```diff
@@ -51,23 +51,10 @@
     }
 
     document.addEventListener('click', (e) => {
-      const path = e.path;
-      const pathLen = path.length;
-      let shouldClose = true;
-
-      for (let j = 0; j < pathLen; j++) {
-        const el = path[j];
-
-        if (el.classList && (el.classList.contains('button-dropdown_content') || el.classList.contains('button-dropdown_icon') || el.classList.contains('button-dropdown_trigger'))) {
-          shouldClose = false;
-          break;
-        }
-      }
-
-      if (shouldClose) {
+      const el = $(e.target);
+      if (!el.closest('.button-dropdown').length && !el.is('.button-dropdown')) {
```

Comment:
"Closest" is built into modern browsers and RB polyfills it for older browsers, so we should only need to refactor `is` into `hasClass`.

Reply:
but hasClass won't work if el is not a jQuery object, right? or does RB already have a polyfill for that?

Reply:
Whoops, got mixed up… meant `classList.contains`. Though you might need to use `classList.indexOf` for IE9 support… double check me. 😛

Reply:
Yes, there's a polyfill for `closest` in the next RB release (which is ready any second now as soon as I write up the changelog). Sub in the [`development` version of `rocketbelt/base/rocketbelt.js`](https://github.com/Pier1/rocketbelt/blob/2516f691c6f6a35cfad632e4e9f3cd73cc6eeeb7/rocketbelt/base/rocketbelt.js#L94) if you'd like to test it out.